### DISCUSSION TITLE
RID Joiner Discussion

### DISCUSSION THREAD

**Discussion by JTD420** [YimMenu/YimMenu#299] [General]
**Is your feature request related to a problem? Please describe.**
ATM there is no way of joining a specific game or session or person that you would like to join. 
The console already logs users names and RID so adding some type of joiner would be a nice addition and be a really powerful feature!

**Describe the solution you'd like**
RID/Player Search/Joiner 

Player History? (keep a history of all players you've been in sessions with recently? (optionally add track/note features like Stand's player tracking features)

**Provide Reasoning**
Sometimes I am helping someone for example and get kicked and with a **P2C** menu I can just rejoin and continue helping out users & kick whoever it was that kicked me but without a way of rejoining sessions or following previously seen users I simply can't do it with just this menu.

**Comment by Yimura** [comment]
1. Please make one issue per request, you're asking for several different things here.
2. One of those several requests already is already an existing issue
3. Why is your title written as if you're trying to be the top result on a search engine?

Check if your request relates to old/closed issues, RID joiner has already been requested here #151.
Session browser is your second request within the same issue which has been requested here #149.

**Comment by xM4ddy** [comment]
This has been discussed serveral times and the last thing I heard was a no on an RID joiner being added to yim menu due to it's possibility of being abused as a toxic feature. Imo the code is already in a discussion on here and in multiple other places, free menus have it, and it has utility so I think it *should* be added but I also understand the argument against it, especially if it includes being able to join invite only sessions.

**Comment by Yimura** [comment]
I will reopen this issue as a place to discuss if an RID joiner should be added to YimMenu.
@JTD420 please edit your issue to only contain the request for RID joiner.

**Comment by Tayipc38** [comment]
A case could be made regarding how useful a RID joiner would be. The fact that this should be permanently implemented in YimMenu does not sit with me considering our general vibe of how we present this menu. Toxicity is a topic we rather evade and this feature would enable a great deal of exactly that. Now I know that in case we made a poll to vote for its implementation the popular vote would be "Yes", so I don't doubt the popularity of this request. 

Alas, I am open to argue whether this feature would be toxic, and how we could mitigate that.

**Comment by xM4ddy** [comment]
> Alas

I like that word it's fun, reminds me of pirates.

Possible mitigation ideas (spitballing)
1) Don't bypass invite only stuff, they are probably alone for a reason.
2) Rate limit it somehow? (doesn't seem like a good method)
3) make it not an RID joiner but a session history joiner or session history and friend rid joiner


There are probably some better ideas just a few of the first thing I could think of.

**Comment by JTD420** [comment]
> 1. Please make one issue per request, you're asking for several different things here.
> 2. One of those several requests already is already an existing issue
> 3. Why is your title written as if you're trying to be the top result on a search engine?
> 
> Check if your request relates to old/closed issues, RID joiner has already been requested here #151. Session browser is your second request within the same issue which has been requested here #149.

Apologies! It was late and I was trying to suggest feature improvements/ideas I had without it seeming spammy to open multiple separate requests. Admittedly should have done a search first!



> I will reopen this issue as a place to discuss if an RID joiner should be added to YimMenu. @JTD420 please edit your issue to only contain the request for RID joiner.

I have edited the OP to be specific to the RID Join feature suggestions/request.

**Comment by JTD420** [comment]
With regards to the discussion, I can conquer with the concerns that may arise from adding this feature without safeguards. That said, I do feel this is a feature that the benefits of adding it outweigh the potential risks/negatives.

My reasoning is this menu, unlike most other mod menus for GTA, specifically tries to avoid adding toxic features, there are no griefing or trolling options etc.

This menu does not cater to the wants or needs of the toxic griefers and trolls.
There are other free menus and plenty of p2c menus which offer them far more options to suit their toxic needs. 

There is not even an argument to be had that if we were to add an RID joiner that the other menus which offer these toxic features could suddenly be made even more toxic by allowing them to use the RID joiner from here because
A) 99.9% of these toxic menus already have their own joiners.
B) You can literally download any number of different free cheats that include a way of joining users
C) There are public external RID joiners available
D) Even public cheat tables exist with this as an option - See: https://github.com/plumbwicked/Wicked-Menu-for-GTA5

So while I can agree with the negative sentiment behind adding such a feature I believe the benefits it provides are far more significant than the less than 1% of users who will use it FROM this menu in a toxic way.

As long as this menu stays true to its goals, avoiding adding any toxic features like crashing, trolling features etc and instead focuses on adding protections, useful and fun features I can't see any reason we would have any problems when Take2 have far worse free and p2cs on their radar to deal with

**Comment by xM4ddy** [comment]
Actually a few toxic options (while not terrible) do exist in yim menu and we have been in discussion about removing them completely. I think a good trade would be to remove those at the same time to also discourage toxic use of the feature if RID join gets added.

**Comment by Yimura** [comment]
Yep sounds good to me tbf.

**Comment by JTD420** [comment]
> Actually a few toxic options (while not terrible) do exist in yim menu and we have been in discussion about removing them completely. I think a good trade would be to remove those at the same time to also discourage toxic use of the feature if RID join gets added.

If you're referring to the explode option then yes I totally agree and support this suggestion!

I noticed it last night while testing controller support and was actually quite surprised it was in there!

If you're referring to the vehicle fly option I think that should be up for discussion too tbh as I can definitely see its clear risk of being abused but I personally have only ever used it for legit reasons such as helping people win the highest free fall challenge or just for fun with friends etc.

Not the end of the world if its removed as I'd far prefer the addition of an RID joiner but if there's a way of keeping both options (possibly preventing it on people who aren't your friends?) then that would be great!

**Comment by Tayipc38** [comment]
> With regards to the discussion, I can conquer with the concerns that may arise from adding this feature without safeguards. That said, I do feel this is a feature that the benefits of adding it outweigh the potential risks/negatives.
> 
> My reasoning is this menu, unlike most other mod menus for GTA, specifically tries to avoid adding toxic features, there are no griefing or trolling options etc.
> 
> This menu does not cater to the wants or needs of the toxic griefers and trolls. There are other free menus and plenty of p2c menus which offer them far more options to suit their toxic needs.
> 
> There is not even an argument to be had that if we were to add an RID joiner that the other menus which offer these toxic features could suddenly be made even more toxic by allowing them to use the RID joiner from here because A) 99.9% of these toxic menus already have their own joiners. B) You can literally download any number of different free cheats that include a way of joining users C) There are public external RID joiners available D) Even public cheat tables exist with this as an option - See: https://github.com/plumbwicked/Wicked-Menu-for-GTA5
> 
> So while I can agree with the negative sentiment behind adding such a feature I believe the benefits it provides are far more significant than the less than 1% of users who will use it FROM this menu in a toxic way.
> 
> As long as this menu stays true to its goals, avoiding adding any toxic features like crashing, trolling features etc and instead focuses on adding protections, useful and fun features I can't see any reason we would have any problems when Take2 have far worse free and p2cs on their radar to deal with

Your point about the feature being readily and freely available is quite convincing. I wasn't worried about adding toxicity to the already misfortunate state of GTA online, it was more the reputation of this open source project. Disregarding my concern, a griefer that's out to grief, will indeed consult to other projects that are easily accessed.
